gem/envs/math_env.py

```diff
@@ -14,19 +14,25 @@
 
 """Env for math datasets."""
 
+import functools
 import logging
 import multiprocessing
 import random
 from typing import Any, Optional, SupportsFloat, Tuple
 
 from datasets import Dataset, DatasetDict, load_dataset
-from math_verify import parse, verify
+from oat.utils.math_grader import boxed_reward_fn
```

Comment:
We'd better not rely on oat dependency. We can copy the code into `gem.utils.math_grader.py` for example. Another way is to let users pass a `verify_fn` when initializing, so `train_oat` can just pass its grader to the env.